application/src/main/java/org/thingsboard/server/service/transport/DefaultTransportApiService.java

```diff
@@ -161,6 +161,8 @@ public ListenableFuture<TbProtoQueueMsg<TransportApiResponseMsg>> handle(TbProto
             result = validateCredentials(msg.getHash(), DeviceCredentialsType.X509_CERTIFICATE);
         } else if (transportApiRequestMsg.hasGetOrCreateDeviceRequestMsg()) {
             result = handle(transportApiRequestMsg.getGetOrCreateDeviceRequestMsg());
+        } else if (transportApiRequestMsg.hasGetOrCreateDeviceSparlplugRequestMsg()) {
```

Comment:
Could we use GetOrCreateDeviceRequestMsg instead of adding a new message type?

Reply:
I agree.

Reply:
Ok